js/views/components/CryptoCurSelector.js

```diff
@@ -82,68 +58,79 @@ export default class extends baseVw {
       activeCurs,
     });
 
-    this.setState({
-      activeCurs,
-    });
+    this.setState({ activeCurs });
   }
 
+
   setState(state = {}, options = {}) {
+    const controlTypes = ['checkbox', 'radio'];
     const curState = this.getState();
 
+    if (state.hasOwnProperty('controlType') &&
+      !controlTypes.includes(state.controlType)) {
+      throw new Error('If provided the controlType must be a valid value.');
+    }
+
+    const checkCurArray = (fieldName => {
+      if (state.hasOwnProperty(fieldName) &&
+        !Array.isArray(state[fieldName])) {
+        throw new Error(`If provided the ${fieldName} must be an array.`);
+      }
+    });
+
+    ['currencies', 'activeCurs', 'disabledCurs']
+      .forEach(field => checkCurArray(field));
+
+    // This is a derived field and should not be directly set
+    delete state.processedCurs;
+
     const processedState = {
+      ...curState,
       ...state,
-      // This is a derived field and should not be directly set
-      processedCurs: Array.isArray(curState.processedCurs) ?
-        curState.processedCurs : [],
+      currencies: Array.isArray(state.currencies) ?
+        [...new Set(state.currencies)] : curState.currencies,
     };
 
-    // De-dupe any passed in currencies.
-    if (state.currencies && Array.isArray(state.currencies)) {
-      processedState.currencies = [...new Set(state.currencies)];
+    // Remove any disabled currencies from the active list.
+    if (state.activeCurs || state.disabledCurs) {
```

Comment:
I started using the pattern a lot here where, rather than using `isEqual` to test if things changed, I just assume if they are passing something in, it's changed.